integrations/gitlab/gitlab_integration/gitlab_service.py

```diff
@@ -533,6 +550,84 @@ async def get_all_issues(self, group: Group) -> typing.AsyncIterator[List[Issue]
             issues: List[Issue] = typing.cast(List[Issue], issues_batch)
             yield issues
 
+    async def get_member_groups(
+        self, member: User
+    ) -> typing.AsyncIterator[List[Group]]:
+
+        semaphore = asyncio.Semaphore(MAX_CONCURRENT_TASKS)
+
+        async def check_group_membership(group: Group) -> Group | None:
+            "check if the user is a member of the group"
+            async with semaphore:
+                try:
+                    await AsyncFetcher.fetch_single(group.members.get, member.get_id())
+                    return group
+                except GitlabError as err:
+                    if err.response_code != 404:
+                        raise err
+                    return None
+
+        async for groups_batch in self.get_all_groups():
+            tasks = [check_group_membership(group) for group in groups_batch]
+            groups = await asyncio.gather(*tasks)
+            member_groups: List[Group] = typing.cast(
+                List[Group], list(filter(None, groups))
+            )
+            logger.info(
+                f"Queried {len(member_groups)} groups {[member_group.name for member_group in member_groups]} for user {member.name}"
+            )
+            yield member_groups
+
+    async def get_all_group_members(
+        self, group: Group
+    ) -> typing.AsyncIterator[List[GroupMemberAll]]:
+
+        logger.info(f"Fetching all members of group {group.name}")
+
+        async for users_batch in AsyncFetcher.fetch_batch(
+            fetch_func=group.members_all.list,
+            validation_func=self.should_run_for_member,
+            pagination="offset",
+            order_by="id",
+            sort="asc",
+        ):
+            members: List[GroupMemberAll] = typing.cast(
+                List[GroupMemberAll], users_batch
+            )
+            logger.info(
+                f"Queried {len(members)} members {[user.username for user in members]} from {group.name}"
+            )
+            yield members
+
+    async def enrich_member_with_groups_and_public_email(
+        self, member
+    ) -> dict[str, Any]:
+        user: User = await self.get_user(member.id)
+
+        user_groups: List[dict[str, Any]] = [
+            {"id": group.id, "full_path": group.full_path}
+            async for groups in self.get_member_groups(user)
+            for group in groups
+        ]
```

Comment:
I am not sure this should be the default behavior when quering members.

Reply:
maybe this should be something that is part of group members kind? that will query the list of members of each group. 
and instead of having a relation of user -> groups, we will have relation between group -> users